### ISSUE TITLE
Unable to use Craft Debug Toolbar

### ISSUE THREAD

**Issue by peteralewis** [born05/craft-csp#3]
Alas once the debug toolbar is turned on, it injects style and scripts inline onto the page. I tried allowing inline and manually adding the toolbar with nonce tags, but best I could achieve was the toolbar appearing but no details beneath (ajax issues with frameSrc?).

Because the tags still need to parsed, I had to add the following around line 67 of Plugin.php to prevent inject of the Content Security Policy headers if the toolbar is visible for the active user:

```
if (Craft::$app->getSession()->get('enableDebugToolbarForSite')) return;
```

Just before `$this->headers->setHeaders();`

Might not be the cleanest way to fix, but all I could come up with.

**Comment by roelvanhintum** [comment]
Thanks! Fixed in 1.2.0

**Comment by peteralewis** [comment]
Sorry, going to have to re-open this one.

I believe Craft v3.5.0 changed something regarding this setting and it appears to no longer be stored in the session:
"Deprecated craft\web\User::destroyDebugPreferencesInSession()."
"Deprecated craft\web\User::saveDebugPreferencesToSession()."

The addition you made to Plugin.php (line 61) to fix this I had to change to:
if ( Craft::$app->users->getUserPreference(Craft::$app->user->id,'enableDebugToolbarForSite') ) return;

Couldn't work out how to do it using the user class - it's documented that getPreference() is available, but doesn't work. So hence having to use users class.
